server/services/interventionsService.test.ts

```diff
@@ -1290,6 +1290,233 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
+                description: 'Attend training course',
+                desiredOutcomeId: '301ead30-30a4-4c7c-8296-2768abfb59b5',
+              },
+            ],
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      })
+
+      const draftActionPlan = await interventionsService.updateDraftActionPlan(token, draftActionPlanId, {
+        activity: {
+          description: 'Attend training course',
+          desiredOutcomeId: '301ead30-30a4-4c7c-8296-2768abfb59b5',
+        },
+      })
+
+      expect(draftActionPlan.id).toBe(draftActionPlanId)
+      expect(draftActionPlan.referralId).toBe('81d754aa-d868-4347-9c0f-50690773014e')
+      expect(draftActionPlan.activities![0].id).toEqual('91e7ceab-74fd-45d8-97c8-ec58844618dd')
+      expect(draftActionPlan.activities![0].description).toEqual('Attend training course')
+      expect(draftActionPlan.activities![0].desiredOutcomeId).toEqual('301ead30-30a4-4c7c-8296-2768abfb59b5')
+    })
+
+    it('updates and returns the newly-updated draft action plan when setting number of sessions', async () => {
+      const draftActionPlanId = 'dfb64747-f658-40e0-a827-87b4b0bdcfed'
+
+      await provider.addInteraction({
+        state: `an action plan exists with id ${draftActionPlanId}`,
+        uponReceiving: 'a PATCH request to set the number of sessions on the action plan',
+        withRequest: {
+          method: 'PATCH',
+          path: `/draft-action-plan/${draftActionPlanId}`,
+          headers: { Accept: 'application/json', Authorization: `Bearer ${token}` },
+          body: {
+            numberOfSessions: 4,
+          },
+        },
+        willRespondWith: {
+          status: 201,
+          body: Matchers.like({
+            id: draftActionPlanId,
+            referralId: '81d754aa-d868-4347-9c0f-50690773014e',
+            numberOfSessions: 4,
+            activities: [
+              {
+                id: '91e7ceab-74fd-45d8-97c8-ec58844618dd',
+                description: 'Attend training course',
+                desiredOutcomeId: '301ead30-30a4-4c7c-8296-2768abfb59b5',
+              },
+              {
+                id: 'e5755c27-2c85-448b-9f6d-e3959ec9c2d0',
+                description: 'Attend session',
+                desiredOutcomeId: '65924ac6-9724-455b-ad30-906936291421',
+              },
+            ],
+            submittedBy: null,
+            createdAt: '2020-12-07T20:45:21.986389Z',
+          }),
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        },
+      })
+
+      const draftActionPlan = await interventionsService.updateDraftActionPlan(token, draftActionPlanId, {
+        numberOfSessions: 4,
+      })
+      expect(draftActionPlan.id).toBe(draftActionPlanId)
+      expect(draftActionPlan.referralId).toBe('81d754aa-d868-4347-9c0f-50690773014e')
+      expect(draftActionPlan.numberOfSessions).toBe(4)
+    })
```

Comment:
Perhaps an activity count of 2?

Reply:
Have added assertions on activity content now